src/components/VmDialog/index.js

```diff
@@ -201,7 +203,13 @@ class VmDialog extends React.Component {
   }
 
   onChangeVmName (event) {
-    this.setState({ name: event.target.value, isChanged: true })
+    var regexp = /^[a-zA-Z0-9._\-+]+$/
```

Comment:
There is a catch there:

Engine uses [`"^[\\p{L}0-9._-]*$"`][1] where `L` refers to [all (Unicode) letters][2] category. However javascript doesn't have any direct ability to to filter unicode categories.

Solutions I can see:
1. Being to strict, current version (probably not an option especially because of users used scripts other than latin)
2. Allow all non-wihitespace chars (probably not much useful)
3. Include a lib like [unicode][3] that contains pretty large [unicode table][4] and allows to do proper validation.
4. Develop custom regexp that would match *L* category - far more space efficient compared to option 3

I'd suggest option 3 - it's easy to do, accurate, and the lib size should be bearable.

[1]: https://github.com/oVirt/ovirt-engine/blob/2c236cae373b5634a73a0d992d523bf3aa551f27/backend/manager/modules/common/src/main/java/org/ovirt/engine/core/common/utils/ValidationUtils.java#L23
[2]: https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#ucc
[3]: https://www.npmjs.com/package/unicode
[4]: https://unicode.org/Public/UNIDATA/UnicodeData.txt

Reply:
what's wrong with 2?

Reply:
It catches just whitespaces. Chars like '!@#$%🚂🚙🚜...' will pass frontend validation but will fail on engine.

Reply:
I like either 1 or 3